packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/representations/EntityNodeDescriptionProvider.java

```diff
@@ -72,22 +89,200 @@ private NodeDescription entityNodeDescription() {
                 .condition("aql:self.abstract")
                 .build();
 
-        var palette = new DiagramBuilders()
-                .newNodePalette()
-                .toolSections(new DefaultToolsFactory().createDefaultHideRevealNodeToolSection())
-                .build();
 
-        return new DiagramBuilders()
+        var entityNodeDescription = new DiagramBuilders()
                 .newNodeDescription()
-                .name("Entity")
+                .name(ENTITY_NODE_DESCRIPTION_NAME)
                 .domainType("domain::Entity")
                 .semanticCandidatesExpression("aql:self.types")
                 .insideLabel(this.entityNodeLabelDescription())
                 .childrenDescriptions(this.attributeNodeDescription())
                 .childrenLayoutStrategy(childrenLayoutStrategy)
                 .style(nodeStyle)
                 .conditionalStyles(conditionalNodeStyle)
-                .palette(palette)
+                .build();
+
+        var palette = this.entityNodePalette(entityNodeDescription);
+        entityNodeDescription.setPalette(palette);
+
+        return entityNodeDescription;
+    }
+
+    private NodePalette entityNodePalette(NodeDescription edgeTargetNodeDescription) {
+        var nodeToolSection = this.entityNodeCreationTools();
+        var relationEdgeTool = this.entityRelationEdgeTool(edgeTargetNodeDescription);
+        var containmentEdgeTool = this.entityContainmentEdgeTool(edgeTargetNodeDescription);
+        var supertypeEdgeTool = this.entitySupertypeEdgeTool(edgeTargetNodeDescription);
+        var labelEditTool = this.labelEditTool();
+        var deleteTool = this.deleteTool();
+        return new DiagramBuilders()
+                .newNodePalette()
+                .labelEditTool(labelEditTool)
+                .deleteTool(deleteTool)
+                .edgeTools(relationEdgeTool, containmentEdgeTool, supertypeEdgeTool)
+                .toolSections(nodeToolSection, new DefaultToolsFactory().createDefaultHideRevealNodeToolSection())
+                .build();
+    }
+
+    private EdgeTool entityRelationEdgeTool(NodeDescription edgeTargetNodeDescription) {
+        return new EdgeToolBuilder()
```

Comment:
Please split those model operations using variables because it's painful to read.